File: app/src/main/java/com/interswitchng/interswitchpossdkdemo/Keyboard.java
package com.interswitchng.interswitchpossdkdemo;

import android.app.Activity;
import android.content.Context;
import android.view.View;
import android.widget.ImageView;
import android.widget.Toast;

import com.interswitchng.smartpos.R;

public class Keyboard implements View.OnClickListener {

    interface KeyBoardListener {
        void onTextChange(String text);

        void onSubmit(String text);
    }

    private Context mContext;
    private String result = "";
    private KeyBoardListener mCallback;


    public Keyboard(Activity activity, KeyBoardListener callback) {
        setupButtons(activity, callback);
    }


    private void setupButtons(Activity activity, KeyBoardListener callback) {
        mContext = activity;
        // add buttons
        activity.findViewById(R.id.one).setOnClickListener(this);
        activity.findViewById(R.id.two).setOnClickListener(this);
        activity.findViewById(R.id.three).setOnClickListener(this);
        activity.findViewById(R.id.four).setOnClickListener(this);
        activity.findViewById(R.id.five).setOnClickListener(this);
        activity.findViewById(R.id.six).setOnClickListener(this);
        activity.findViewById(R.id.seven).setOnClickListener(this);
        activity.findViewById(R.id.eight).setOnClickListener(this);
        activity.findViewById(R.id.nine).setOnClickListener(this);
        activity.findViewById(R.id.oneZero).setOnClickListener(this);
        activity.findViewById(R.id.twoZeros).setOnClickListener(this);
        activity.findViewById(R.id.threeZeros).setOnClickListener(this);
        activity.findViewById(R.id.done).setOnClickListener(this);

        ImageView delete = activity.findViewById(R.id.delete);
        delete.setOnClickListener(this);
        // clear text on long click
        delete.setOnLongClickListener(new View.OnLongClickListener() {
            @Override
            public boolean onLongClick(View v) {
                result = "";
                if (mCallback != null) mCallback.onTextChange(result);
                return true;
            }
        });


        // set callback
        mCallback = callback;
    }

    @Override
    public void onClick(View view) {
        long maxValue = 1_000_000_00L; // 1 million (1,000,000.00)
        String result = this.result;

        switch (view.getId()) {
            case R.id.one: {
                result += "1";
                break;
            }
            case R.id.two: {
                result += "2";
                break;
            }
            case R.id.three: {
                result += "3";
                break;
            }
            case R.id.four: {
                result += "4";
                break;
            }
            case R.id.five: {
                result += "5";
                break;
            }
            case R.id.six: {
                result += "6";
                break;
            }
            case R.id.seven: {
                result += "7";
                break;
            }
            case R.id.eight: {
                result += "8";
                break;
            }
            case R.id.nine: {
                result += "9";
                break;
            }
            case R.id.oneZero: {
                result += "0";
                break;
            }
            case R.id.twoZeros: {
                result += "00";
                break;
            }
            case R.id.threeZeros: {
                result += "000";
                break;
            }
            case R.id.delete: {
                if (result.length() > 0) {
                    result = result.substring(0, result.length() - 1);
                }
                break;
            }
            case R.id.done: {
                if (mCallback != null) mCallback.onSubmit(result);
            }
            default: {
                return;
            }
        }

        boolean isDisabled = !result.isEmpty() && (Long.valueOf(result) > maxValue);

        if (!isDisabled) {
            this.result = result;
            if (mCallback != null) mCallback.onTextChange(result);
        } else {
            // show max input
            Toast.makeText(mContext, "Max value is 1 Million", Toast.LENGTH_SHORT).show();
        }

    }

    public void setText(String text) {
        result = text;
    }
}

File: smart-pos-core/src/main/java/com/interswitchng/smartpos/modules/home/Keyboard.kt
package com.interswitchng.smartpos.modules.home

import android.app.Activity
import android.content.Context
import android.view.View
import android.widget.ImageView
import android.widget.Toast
import com.interswitchng.smartpos.R


internal class Keyboard(activity: Activity, callback: KeyBoardListener) : View.OnClickListener {

    private var mContext: Context? = null
    private var result = ""
    private var mCallback: KeyBoardListener? = null

    interface KeyBoardListener {
        fun onTextChange(text: String)

        fun onSubmit(text: String)
    }


    init {
        setupButtons(activity, callback)
    }


    private fun setupButtons(activity: Activity, callback: KeyBoardListener) {
        mContext = activity
        // add buttons
        activity.findViewById<View>(R.id.one).setOnClickListener(this)
        activity.findViewById<View>(R.id.two).setOnClickListener(this)
        activity.findViewById<View>(R.id.three).setOnClickListener(this)
        activity.findViewById<View>(R.id.four).setOnClickListener(this)
        activity.findViewById<View>(R.id.five).setOnClickListener(this)
        activity.findViewById<View>(R.id.six).setOnClickListener(this)
        activity.findViewById<View>(R.id.seven).setOnClickListener(this)
        activity.findViewById<View>(R.id.eight).setOnClickListener(this)
        activity.findViewById<View>(R.id.nine).setOnClickListener(this)
        activity.findViewById<View>(R.id.oneZero).setOnClickListener(this)
        activity.findViewById<View>(R.id.twoZeros).setOnClickListener(this)
        activity.findViewById<View>(R.id.threeZeros).setOnClickListener(this)
        activity.findViewById<View>(R.id.done).setOnClickListener(this)

        val delete = activity.findViewById<ImageView>(R.id.delete)
        delete.setOnClickListener(this)
        // clear text on long click
        delete.setOnLongClickListener { _ ->
            result = ""
            if (mCallback != null) mCallback!!.onTextChange(result)
            true
        }


        // set callback
        mCallback = callback
    }

    override fun onClick(view: View) {
        val maxValue = 1_000_000_00L // 1 million (1,000,000.00)
        var result = this.result

         when (view.id) {
            R.id.one        -> result += "1"
            R.id.two        -> result += "2"
            R.id.three      -> result += "3"
            R.id.four       -> result += "4"
            R.id.five       -> result += "5"
            R.id.six        -> result += "6"
            R.id.seven      -> result += "7"
            R.id.eight      -> result += "8"
            R.id.nine       -> result += "9"
            R.id.oneZero    -> result += "0"
            R.id.twoZeros   -> result += "00"
            R.id.threeZeros -> result += "000"
            R.id.done       -> mCallback?.onSubmit(result)
            R.id.delete     -> if (result.isNotEmpty()) {
                result = result.substring(0, result.length - 1)
            }
            else -> return
        }

        val isDisabled = !result.isEmpty() && result.toLong() > maxValue

        if (!isDisabled) {
            this.result = result
            if (mCallback != null) mCallback!!.onTextChange(result)
        } else {
            // show max input
            Toast.makeText(mContext, "Max value is 1 Million", Toast.LENGTH_SHORT).show()
        }

    }

    fun setText(text: String) {
        result = text
    }
}
